pkg/types/params.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright © 2021 Cisco and/or its affiliates. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import "fmt"
+
+type Params map[string][]string
```

Comment:
It has been talked before to move the `properties` package from koperator to a separate one. Seems like some of the functionalities has been reimplemented here. Wouldn't it make sense to use that package here?

Reply:
The `properties` pkg is not relevant here, therefore I prefer not to pull it in. Only the struct tag handling logic is duplicated between the `properties` pkg and this one.

Reply:
Discussed offline. Good to go